signac/contrib/project.py

```diff
@@ -732,63 +729,49 @@ def _find_job_ids(self, filter=None, doc_filter=None):
         ValueError
             If the filters are invalid.
 
-        Notes
-        -----
-        If all filter arguments are empty or ``None``, this method simply
-        returns a list of all job directories. This code path can be much faster
-        for certain use cases since it defers all work that would be required to
-        construct an index. In performance-critical applications where no
-        filtering of the data space is required, passing empty filters (or
-        ``None``) to this method is recommended.
-
         """
-        if not filter and not doc_filter:
+        if not filter:
             return list(self._job_dirs())
         filter = dict(parse_filter(_add_prefix("sp.", filter)))
-        if doc_filter:
-            warnings.warn(DOC_FILTER_WARNING, DeprecationWarning)
-            filter.update(parse_filter(_add_prefix("doc.", doc_filter)))
-            index = self._index(include_job_document=True)
-        elif "doc" in _root_keys(filter):
+        if "doc" in _root_keys(filter):
             index = self._index(include_job_document=True)
         else:
             index = self._sp_index()
-        return Collection(index, _trust=True)._find(filter)
+        return list(Collection(index, _trust=True)._find(filter))
 
-    def find_jobs(self, filter=None, doc_filter=None):
+    def find_jobs(self, filter=None, *args, **kwargs):
         """Find all jobs in the project's workspace.
 
-        The optional filter arguments must be a Mapping of key-value pairs and
-        JSON serializable. The ``filter`` argument is used to search against job
-        state points, whereas the ``doc_filter`` argument compares against job
-        document keys.
+        The filter argument must be a JSON-serializable Mapping of key-value
+        pairs. The ``filter`` argument can search against both job state points
+        and job documents. See
+        https://docs.signac.io/en/latest/query.html#query-namespaces
+        for a description of supported queries.
 
         See :ref:`signac find <signac-cli-find>` for the command line equivalent.
 
         Parameters
         ----------
         filter : Mapping
```

Comment:
Is `Mapping` referring here to the ABC? That might be misleading.

Reply:
We do support anything adhering to the abstract base class `Mapping`. It is correct.

Reply:
Ok.